src/engine/Server.cpp

```diff
@@ -337,20 +367,17 @@ Awaitable<void> Server::process(
       throw std::runtime_error(
           "Parameter \"query\" must not have an empty value");
     }
-    std::chrono::seconds timeLimit =
-        RuntimeParameters().get<"default-query-timeout">();
-    if (auto timeoutParam = checkParameter("timeout", std::nullopt)) {
-      std::string userTimeoutString{timeoutParam.value()};
-      // std::stoi does not support std::string_view
-      std::chrono::seconds userTimeout{std::stoi(userTimeoutString)};
-      if (userTimeout > timeLimit) {
-        // Throw exception if token is invalid
-        checkParameter("timeout", std::nullopt, accessTokenOk);
-      }
-      timeLimit = userTimeout;
+
+    if (auto timeLimit = co_await verifyUserSubmittedQueryTimeout(
+            checkParameter("timeout", std::nullopt), accessTokenOk, request,
+            send)) {
+      co_return co_await processQuery(parameters, requestTimer,
+                                      std::move(request), send,
+                                      timeLimit.value());
+
+    } else {
```

Comment:
Add a comment here that the `verify...`  has already sent an error response if it returns `nullopt`, otherwise this code here looks a little bit strange.